client_sdk/go/fpc/management.go

```diff
@@ -71,58 +65,40 @@ func (c *managementState) InitEnclave(peerEndpoint string, attestationParams ...
 		return err
 	}
 
-	// Set attestation paramaters
-	type Params struct {
-		AttestationType string `json:"attestation_type"`
-		HexSpid         string `json:"hex_spid"`
-		SigRL           string `json:"sig_rl"`
+	// TODO parse attestationParams
+	// Currently we ignore the function inputs and read the attestation params from environment, needs to be fixed
```

Comment:
I think the best way would be to define the attestationParams as optional (e.g., by having a separate attestationParam-less version of `InitEnclave`, with this one calling `CreateAttestationParamsFromEnvironment` and then calling the version with the explicit arg?)

Reply:
Actually, i just realize that i'm a bit confused. Isn't this function essentially now a private one as folks should use the LifecycleInitEnclave one?  Having it public would seem somewhat confusing?

And from that perspective, i think also above suggestion doesn't make sense.  Seems to me we actually should handle that via `LifecycleInitEnclaveRequest ` and handle it similar to `SGX_MODE`. There is a field which when defined is used from the request and if not defined we read it from the environment?

Reply:
well, I guess the `InitEnclave` command in management.go will disappear and is replaced with the `LifecycleInitEnclave` command. 

With the new `LifecycleInitEnclave` the attestation params are passed via `LifecycleInitEnclaveRequest`. So the user is responsible to gather the attestation params. For convenience we can provide methods like ...

```go
func CreateAttestationParamsFromEnvironment() (*AttestationParams, error) {}
func CreateAttestationParamsFromCredentialsPath(sgxCredentialsPath string) (*AttestationParams, error) {}
...
```

Reply:
> well, I guess the `InitEnclave` command in management.go will disappear and is replaced with the `LifecycleInitEnclave` command.

I thought you actually still call `InitEnclave` from `LifecycleInitEnclave` in which case it couldn't really disappear?  That said, i don't really care about either disappearing or being private as long as it wouldn't be visible in the public interface/godoc anymore :-)

> With the new `LifecycleInitEnclave` the attestation params are passed via `LifecycleInitEnclaveRequest`. So the user is responsible to gather the attestation params. For convenience we can provide methods like ...
> 
> ```go
> func CreateAttestationParamsFromEnvironment() (*AttestationParams, error) {}
> func CreateAttestationParamsFromCredentialsPath(sgxCredentialsPath string) (*AttestationParams, error) {}
> ...
> ```

I thought you already had some public functions to get them (parts of it) from environment?   In any case, my proposal to them optional parameter which if absent would be automatically filled from environment was just a thought; i'm fine to always require them in the request object ...

Reply:
> I thought you actually still call InitEnclave from LifecycleInitEnclave in which case it couldn't really disappear? 

no, I am not calling the `initEnclave` from mangement.go as this is based on the gateway API which does not interact nicely with the "other" sdk clients such as channel or resmgmt client.

> That said, i don't really care about either disappearing or being private as long as it wouldn't be visible in the public interface/godoc anymore :-)
Sure, I still need to do that.